stdlib/3/dis.pyi

```diff
@@ -0,0 +1,70 @@
+from typing import Any, List, Union, NamedTuple, Iterator, Tuple, IO, Optional, Dict
+
+from opcode import cmp_op,hasconst ,hasname , hasjrel, hasjabs, haslocal, hascompare, hasfree, opname ,opmap ,HAVE_ARGUMENT, EXTENDED_ARG, hasnargs
+
+import types
+
+_have_code = Union[types.MethodType, types.FunctionType, types.CodeType, type]
+_have_code_or_string = Union[str, bytes, types.MethodType, types.FunctionType, types.CodeType, type]
+
+class Instruction(NamedTuple("Instruction", [
+        ('opname', str),
+        ('opcode', int),
+        ('arg', Optional[int]),
+        ('argval', Any),
+        ('argrepr', str),
+        ('offset', int),
+        ('starts_line', Optional[int]),
+        ('is_jump_target', bool)
+    ])):
+    # ad-hoc - seems to be an error in the NamedTuple type hint
+    def _replace(self, **kwargs:Any) -> Instruction: ...
+    def _fields(self) -> List[str]: ...
+
+
+import sys
+if sys.version_info >= (3, 4):
+    class Bytecode:
+        codeobj = ... # type: types.CodeType
+        first_line = ... # type: int
+        def __init__(self, x:_have_code_or_string, *, first_line:int=..., current_offset:int=...) -> None: ...
+        def __iter__(self) -> Iterator[Instruction]: ...
+        def __repr__(self) -> str: ...
+        def info(self) -> str: ...
+        def dis(self) -> str: ...
+    
+        @classmethod
+        def from_traceback(cls, tb:types.TracebackType) -> Bytecode: ...
+ 
+COMPILER_FLAG_NAMES = ... # type: Dict[int, str]
+
+
+def pretty_flags(flags:int) -> str: ...
+def findlabels(code:_have_code) -> List[int]: ...
+def findlinestarts(code:_have_code) -> Iterator[Tuple[int, int]]: ...
+
+if sys.version_info < (3, 4):
```

Comment:
All the version checks are confusing, especially since mypy doesn't support them yet. Maybe it would be better to just put the Python 3.5 API in the stub, with TODO comments about which parameters/functions/fields aren't supported in earlier Python versions?